runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineRunner.java

```diff
@@ -115,13 +114,6 @@ public PortablePipelineResult run(RunnerApi.Pipeline pipeline, JobInfo jobInfo)
             : GreedyPipelineFuser.fuse(trimmedPipeline).toPipeline();
 
     // File staging.
-    if (pipelineOptions.getFilesToStage() == null) {
-      pipelineOptions.setFilesToStage(
-          detectClassPathResourcesToStage(
-              SparkPipelineRunner.class.getClassLoader(), pipelineOptions));
-      LOG.info(
-          "PipelineOptions.filesToStage was not specified. Defaulting to files from the classpath");
-    }
```

Comment:
I think removing this broke Python PostCommits, see BEAM-12207. We're getting an NPE from the LOG statement below, presumably because `pipelineOptions.getFilesToStage() == null`. I'm not sure what the appropriate fix is, is it expected that now the null checking happens somewhere else, or should we add a null check back here?

Reply:
Fixed by @ibzib via #14616